src/txDigester/api.ts

```diff
@@ -0,0 +1,30 @@
+import { FastifyInstance, FastifyRequest } from 'fastify'
+import { Server, IncomingMessage, ServerResponse } from 'http'
+import { getTxDigestsForACycleRange } from './txDigestFunctions'
+
+type GetTxDigestsRequest = FastifyRequest<{
+  Querystring: {
+    cycleStart: number
+    cycleEnd: number
+  }
+}>
+
+/* To-Do: Add LRU cache for the tx-digests */
+export function registerRoutes(server: FastifyInstance<Server, IncomingMessage, ServerResponse>): void {
+  server.get('/api/tx-digests', async (_request: GetTxDigestsRequest, reply) => {
+    const cycleStart = Number(_request.query?.cycleStart)
+    const cycleEnd = Number(_request.query?.cycleEnd)
+
+    if (isNaN(cycleStart) || isNaN(cycleEnd) || cycleEnd <= cycleStart || cycleStart < 0 || cycleEnd < 0) {
```

Comment:
Let's add one more check, diff between cycleEnd and cycleStart should be <=10000 (so we fetch max 1000 records in a call assuming cycleDiff will never be less than 10).

Reply:
Done